### DISCUSSION TITLE
2 ctrl+backspace scenarios I'd like to (script|configure-enable)

### DISCUSSION THREAD

**Discussion by metaleap** [orbitalquark/textadept#147] [Q&A] [answered]
To smoothen my (or anyone's) transition from the vscode/sublime set, I'd want to either script or configure these 2 ctrl+bksp always-on behaviors:

- the situation: in an indented line, the caret is right at the first non-space char (aka to its left)
  - on ctrl+backspace, would wanna be at the 0 column (1st) of _this_ line, not the end of the previous line as it is now (at my end).
- the situation: caret is at EOL, many spaces apart from the line's actual last non-space char:
  - on ctrl+backspace, would want deletion of just those spaces not the last word itself, so that end-of-word and EOL and caret pos are now the same

For all my exhaustive docs-studying I can't quite reason out how I would induce such... anyone have a clever scripting idea per-chance? Thanks  =)

**Comment by orbitalquark** [comment, marked as answer]
Something like this in your *~/.textadept/init.lua* might work (untested):

keys['ctrl+\b'] = function()
  local pos = buffer.current_pos
  local line = buffer:line_from_position(pos)
  if pos == buffer.line_indent_position[line] then
    buffer:del_line_left()
  elseif pos == buffer.line_end_position[line] then
    buffer:del_line_right()
  end
end

**Reply by metaleap** [reply]
That's the stuff! Merci!
